src/Microsoft.AspNet.Mvc.Core/Description/DefaultApiDescriptionProvider.cs

```diff
@@ -414,6 +421,40 @@ private class ApiParameterContext
             public IReadOnlyList<TemplatePart> RouteParameters { get; }
         }
 
+        private class ApiDescriptorBindingContext
+        {
+            public ModelMetadata ModelMetadata { get; set; }
+
+            public string BinderModelName { get; set; }
+
+            public BindingSource BindingSource { get; set; }
+
+            public Func<ModelBindingContext, string, bool> PropertyFilter { get; set; }
+
+            public Type BinderType { get; set; }
+
+            public string PropertyName { get; set; }
+
+            public static ApiDescriptorBindingContext GetContext(
+                ModelMetadata metadata,
+                BindingInfo bindingInfo,
+                string propertyName)
+            {
+                // BindingMetadata can be null if the metadata represents properties.
+                var propertyPredicateProvider =
+                    bindingInfo?.PropertyBindingPredicateProvider ?? metadata.PropertyBindingPredicateProvider;
+                return new ApiDescriptorBindingContext
+                {
+                    ModelMetadata = metadata,
+                    BinderModelName = bindingInfo?.BinderModelName ?? metadata.BinderModelName,
+                    BindingSource = bindingInfo?.BindingSource ?? metadata.BindingSource,
+                    PropertyFilter = propertyPredicateProvider?.PropertyFilter,
+                    BinderType = bindingInfo?.BinderType ?? metadata.BinderType,
+                    PropertyName = propertyName ?? metadata.PropertyName
+                };
+            }
+        }
```

Comment:
This seems much bigger than what we need. We really only use the source, metadata, and name right?